File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio8/Java/EncriptarFrase.java
import java.util.Scanner;

public class EncriptarFrase {
    public static String encriptarFrase(String frase) {
        String vocales = "aeiouAEIOU";
        StringBuilder fraseEncriptada = new StringBuilder();
        char vocalAnterior = ' ';
        for (char caracter : frase.toCharArray()) {
            if (vocales.indexOf(caracter) != -1) {
                int indiceVocal = vocales.indexOf(caracter);
                if (indiceVocal > 0) {
                    char caracterEncriptado = vocales.charAt(indiceVocal - 1);
                    fraseEncriptada.append(caracterEncriptado);
                } else {
                    fraseEncriptada.append(caracter);
                }
            } else {
                fraseEncriptada.append(caracter);
            }
        }
        return fraseEncriptada.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese una frase: ");
        String frase = scanner.nextLine();
        String fraseEncriptada = encriptarFrase(frase);
        System.out.println("Frase original: " + frase);
        System.out.println("Frase encriptada: " + fraseEncriptada);
        scanner.close();
    }
}

File: PRÁCTICA SEGUNDO PARCIAL/Ejercicio8/Python/EncriptarFrase.py
def encriptar_frase(frase):
    vocales = "aeiouAEIOU"
    frase_encriptada = ""
    vocal_anterior = ""
    for caracter in frase:
        if caracter in vocales:
            indice_vocal = vocales.index(caracter)
            if indice_vocal > 0:
                caracter_encriptado = vocales[indice_vocal - 1]
                frase_encriptada += caracter_encriptado
            else:
                frase_encriptada += caracter
        else:
            frase_encriptada += caracter
    return frase_encriptada

frase = input("Ingrese una frase: ")
frase_encriptada = encriptar_frase(frase)
print("Frase original:", frase)
print("Frase encriptada:", frase_encriptada)
